.travis.yml.nix

```diff
@@ -35,13 +37,42 @@ let
       script = ''
         set -e
         source ./.travis_fold.sh
-        travis_fold ci_check \
+        lorri_travis_fold ci_check \
           nix-shell --quiet --run ci_check
-        travis_fold travis-yml-gen \
+        lorri_travis_fold travis-yml-gen \
           cat $(nix-build --quiet ./.travis.yml.nix --no-out-link) > .travis.yml
-        travis_fold travis-yml-idempotent \
+        lorri_travis_fold travis-yml-idempotent \
           git diff -q ./.travis.yml
       '';
+      # delete all our own artifacts from the cache dir
+      # based on https://gist.github.com/jkcclemens/000456ca646bd502cac0dbddcb8fa307
+    };
+
+    # cache rust dependency building
+    cache = name: {
+      before_cache =
+        let rmTarget = path: "rm -rvf $TRAVIS_BUILD_DIR/target/debug/${path}";
+        in (map rmTarget [
+          "lib${projectname}.rlib"
+          # our own binaries/libraries (keep all other deps)
+          "lorri*"
+          "build/${projectname}-*"
+          "deps/${projectname}-*"
+          "deps/lib${projectname}-*"
+          "incremental/${projectname}-*"
+          ".fingerprint/${projectname}-*"
+          # build script executable
+          "incremental/build_script_build-*"
+          # TODO: the direnv integration test is not deterministic
+          "direnv-*"
+          "deps/direnv-*"
+          "incremental/direnv-*"
```

Comment:
Will it be a problem when this list goes stale?

Reply:
It will cache more things than it needs to. But unless `cargo` changes its `target/` directory structure, I doubt much will change here.